### PR TITLE
[UST-277] [Circuit] Check if reputation is negative and the epoch last claimed proof < 1 day

### LINKED ISSUE #580: [UST-277] [Circuit] Check if reputation is negative and the epoch last claimed proof < 1 day
**Is your feature request related to a problem? Please describe.**
目前還沒有機制去防止用戶一天只能 check in 一次，也沒有檢查用戶 reputation 是否為 negative

**Describe the solution you'd like**
寫一個 dailyClaimTemplate，裡面引入 reputation circuit  再多一個`hash(identity, dailyEpoch)` 當作 public signals作為claim daily reputation 的 nullifier， circuit內需要驗證

- reputation proof 是否正確
- 用戶的reputation < 0
- daily claim nullifier 是否正確

public signals需要有以下這些內容

- reputation proof原有的public signals
- daily epoch
- daily nullifier

```
template DailyClaimProof(STATE_TREE_DEPTH, FIELD_COUNT) {
    signal input identity;
    signal input daily_epoch;

    // TODO: reputation circuit input
    
    /* Step 1. Check reputation circuit */
    
    /* Step 2. Check reputation < 0 */

    /* Step 3: check daily nullifier */
    
}
```

```bash
uint startDate = 1514764800; // 2018-01-01 00:00:00
uint endDate = 1518220800; // 2018-02-10 00:00:00

uint diff = (endDate - startDate) / 60 / 60 / 24; // 40 days 

hash(identity, dailyEpoch) --> nullifier 存進 map
```

- 撰寫測試
    - 通過測試
        - reputation < 0 且 daily epoch 為當天的 epoch
    - 失敗測試
        - reputation > 0
        - daily nullifier 錯誤
        - reputation proof 是錯誤的

### PR DESCRIPTION
**Pull request by chentihe**
## Summary

新增 dailyClaimTemplate，在此circuit內驗證

- reputation proof 是否正確
- 用戶的reputation < 0
- daily claim nullifier 是否正確
- 
public signals需要有以下這些內容，以利後端跟合約進行檢查
- dailyEpoch
- dailyNullifier
- attesterId
- epoch
- minRep
- maxRep

## Linked Issue

close #580 

## Tests
     * 1. should succeed with valid inputs
     * 2. should revert with reputation > 0
     * 3. should revert with wrong reputation proof
     * 4. should revert with wrong daily nullifier

## Checklist

-   [x] All new and existing tests pass
-   [x] I have updated the documentation (if applicable)
-   [x] My changes do not introduce new security vulnerabilities

### CODE REVIEW

**Comment by cc03668** [comment]
@chentihe 這個 PR 名稱請遵守規範做修改唷

File: packages/circuits/test/proveDailyClaim.test.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+const circuit = UnirepSocialCircuit.dailyClaimProof
+
+describe('Prove report identity in Unirep Social-TW', function () {
+    this.timeout(300000)
+    /**
+     * 1. should succeed with valid inputs
+     * 2. should revert with reputation > 0
+     * 3. should revert with wrong reputation proof
+     * 4. should revert with wrong daily nullifier
```
Comment (FoodChain1028) [inline packages/circuits/test/proveDailyClaim.test.ts:25]: Can we add other tests to test: when user get a reputation after claiming a daily rep, and the user claim again within 24 hour?

And I don't know how do you check if **the epoch last claimed** is < 1 day. (I didn't see the check in circuit).

**Review by FoodChain1028 — CHANGES_REQUESTED** [review]
I didn't see where the circuit check if the last epoch claimed is less than 24 hours :(. Maybe I just miss it. Could you point it out where it is? Thanks!

**Review by FoodChain1028 — APPROVED** [review]
LGTM. Just make sure the `1 day check` test will be included in the contract part, thanks!

**Pull request merged**